File: samples/server/petstore/springboot-virtualan/src/main/java/org/openapitools/configuration/HomeController.java
package org.openapitools.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.GetMapping;

/**
 * Home redirection to OpenAPI api documentation
 */
@Controller
public class HomeController {

    @GetMapping(value = "/swagger-config.yaml", produces = "text/plain")
    @ResponseBody
    public String swaggerConfig() {
        return "url: /v2/api-docs\n";
    }
    @RequestMapping("/")
    public String index() {
        return "redirect:swagger-ui.html";
    }

}

File: samples/openapi3/kotlin-spring-boot-bigdecimal-default/src/main/kotlin/org/openapitools/HomeController.kt
package org.openapitools

import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.GetMapping

/**
 * Home redirection to OpenAPI api documentation
 */
@Controller
class HomeController {

    @RequestMapping("/")
    fun index(): String = "redirect:swagger-ui.html"
}
